subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -662,7 +662,7 @@ static int mod_sub_status(struct bt_mesh_model *model,
 		    (param->expect_sub && *param->expect_sub != sub_addr) ||
 		    param->cid != cid) {
 			BT_WARN("Model Subscription Status parameters did not match");
-			return -ENOENT;
+			goto done;
```

Comment:
Hm. It changes behavior. If ack context doesn't match expected the function returns success.

Reply:
When sub command sending is set to non blocking, param will be released immediately. When the status is returned later, the param here is empty. I think the result should be passed to the user, and the user can decide whether to keep it.

Reply:
Please move this callback to [line 658](https://github.com/zephyrproject-rtos/zephyr/blob/950c5f0f32fb6104a456a457e1e823ab44bf36cb/subsys/bluetooth/mesh/cfg_cli.c#L658).

Reply:
> When sub command sending is set to non blocking, param will be released immediately. When the status is returned later, the param here is empty. I think the result should be passed to the user, and the user can decide whether to keep it.

I'm not sure I got your statement. My concern is that if I didn't initialize callback and the context doesn't match application gets Success but actually, nothing happened. Application got stuck forever in the worst case.